### PR TITLE
chore: introduce new (identical) SubNamespace API version

### PR DESCRIPTION
**Pull request by erikgb**
As suggested in https://github.com/cybozu-go/accurate/pull/96, I think it makes a lot of sense to first introduce an identical and unserved API version. Since there must be only one storage version of a CRD in Kubernetes, the current `v1` version is marked as the storage version.

### CODE REVIEW

File: api/v2alpha1/subnamespace_types.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations are the annotations to be propagated to the sub-namespace.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Keeping this version un-served for now
+//+kubebuilder:unservedversion
+//+kubebuilder:object:root=true
```
Comment (ymmt2005) [inline api/v2alpha1/subnamespace_types.go:32]: I guess we're going to make the `status` a subresource.
```suggestion
// Keeping this version un-served for now
// +kubebuilder:unservedversion
// +kubebuilder:object:root=true
// +kubebuilder:subresource:status
```

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

File: api/v2alpha1/subnamespace_types.go (hunk 1 above)
Comment (erikgb) [reply api/v2alpha1/subnamespace_types.go:32]: I was planning to start with an _identical_ new version, but we are heading for a status subresource. Do you prefer heading for https://github.com/cybozu-go/accurate/pull/96 directly?

Comment (ymmt2005) [reply api/v2alpha1/subnamespace_types.go:32]: I understand.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**